### ISSUE TITLE
sed: can't read common/cras_messages.h: no such file or directory

### ISSUE THREAD

**Issue by reboun** [dnschneid/crouton#4946]
#### Result of 'sudo edit-chroot -all':
```
name: xenial
encrypted: no
Entering /mnt/stateful_partition/crouton/chroots/xenial...
crouton: version 1-20221126232257~master:ae594340
release: xenial
architecture: amd64
targets: xfce
host: version 15373.0.0 (Official Build) dev-channel octopus 
kernel: Linux localhost 4.14.254-20300-g04f9d4e19110 #1 SMP PREEMPT Sat Mar 4 20:21:26 PST 2023 x86_64 x86_64 x86_64 GNU/Linux
freon: yes
Unmounting /mnt/stateful_partition/crouton/chroots/xenial...
```
#### I keep having this issue and I cannot install a chroot:
```
sed: can't read common/cras_messages.h: No such file or directory
Failed to complete chroot setup.
Unmounting /mnt/stateful_partition/crouton/chroots/xenial...
```

**Comment by Qwer-TeX** [comment]
I have the same issue. I'm pretty sure some people have this issue. I may update you if I can get it to be fixed.

**Comment by GamerX999999999** [comment]
Same problem. I tried to install bionic and it gave me the same error
`sed: can't read common/cras_messages.h: No such file or directory
Failed to complete chroot setup.
Unmounting /mnt/stateful_partition/crouton/chroots/bionic...
Sending SIGTERM to processes under /mnt/stateful_partition/crouton/chroots/bionic...`

**Comment by drinkcat** [comment]
Yeah, the header files moved to `include`. There'll be bigger problems down the line here, that _could_ be fixed by:
https://github.com/dnschneid/crouton/pull/4959

I wonder if we can just extract the `src/include` directory in `src/cras/common` and hope it works...

**Comment by drinkcat** [comment]
https://chromium-review.googlesource.com/c/chromiumos/third_party/adhd/+/4242301 this broke it.

**Comment by drinkcat** [comment]
You could try this PR, second commit might fix it: https://github.com/dnschneid/crouton/pull/4959/commits/0d48e401d7375724828adc448568c192f4d119dd

(none of this is tested -- at all)

**Comment by NateTheClickbait** [comment]
I also got the software today and the code looks like it will work but I don't know how to use the code or " build it from the source" I would appreciate any help I can get.

**Comment by francorv961** [comment]
Hi guys! I have a similar problem. I have a Chromebook and I want to install ubuntu with xfce, but when I want to install crouton I have the next message:
sed: can't read common/cras_messages.h: No such file or directory
Failed to complete chroot setup.
I hope you can help me. Thanks

**Comment by calvinxp** [comment]
I have the same issue

sed: can't read common/cras_messages.h: No such file or directory
Failed to complete chroot setup.
Unmounting /mnt/stateful_partition/crouton/chroots/xenial...

**Comment by NateTheClickbait** [comment]
Who knows how to inject his code into crouton?

**Comment by jcclin** [comment]
I am having the same issue.  In terms of targets, I tried xenial, focal & jammy, all bumped into the same problem.  
Note the ChromeOS version is 112.

**Comment by zhao-nine** [comment]
hi!
I have same issue in chrome book  to build  crouton.
'sed: can't read common/cras_messages.h: No such file or directory
Failed to complete chroot setup.'
What is the solution to this problem? How to deal with this error?

**Comment by ruitian37** [comment]
I have the same issue
sed: can't read common/cras_messages.h: No such file or directory
Failed to complete chroot setup.
Unmounting /mnt/stateful_partition/crouton/chroots/

**Comment by daywalker247** [comment]
same here..

**Comment by jcclin** [comment]
Thanks @jake01756 

I can confirm 
    `sudo CROUTON_BRANCH=silence crouton -r jammy -t xfce,xiwi`
works for me.
I understand audio is unlikely to work but I am not depending on it.
(Asus Chromebook C204e, ChromeOS version 113.0.5672.114)

**Comment by TangerineTopaz** [comment]
Wait, I'm still confused. @jake01756 At what point in the set up are we running this? I ran into the error, and while I'm still in the shell, I tried running it, but "sudo CROUTON_BRANCH=silence crouton" just pulled up the normal install menu.

How do I know what things to add to the end of it? I'm trying to download xfce for Crouton too.

**Comment by TangerineTopaz** [comment]
Edit: @jcclin 's suggestion worked - just... enter in said command while in shell (no need to clear out last attempt). Audio's definitely fried, but hey, it works! Wonder if there's a work around for sound?

(Considering how janky this is I'm assuming no, and various other distros already seem to have audio issues when it comes to running on Chromebooks).

**Comment by Qwer-TeX** [comment]
` sudo CROUTON_BRANCH=silence crouton` For now lets still get the job done without the audio and with the headphones. 
But try installing sound drivers for chromebook and see if it works. The sound coming out of the speakers sounds like wood burning in a fire in a camp

**Comment by Gorvetco** [comment]
_

> I tried running it, but "sudo CROUTON_BRANCH=silence crouton" just pulled up the normal install menu. 
> How do I know what things to add to the end of it? I'm trying to download xfce for Crouton too.

_ 
So, its fairly straight forward actually. If you've already made a chroot that is currently incomplete, (and haven't specified a name for it (-n) then inevitably it will be namedafter the distro you chose to install. So all you'd need to do now at this point would be to "update" your chroot using the "-u" variable. So lets say the distro you chose was **xenial** and you said you wanted the xfce desktop environment. So therefore your command would look like this:

`sudo CROUTON_BRANCH=silence crouton -u -r xenial -t xfce`

Or lets say you deleted your chroot and you wanted to start a brand new one. You can do the same thing pretty much, only you'd drop the "-u", since you won't be updating and already existing chroot. So in this case your command would look like this:

`sudo CROUTON_BRANCH=silence crouton -r xenial -t xfce`

So remember, you can interchange **xenial** with the distro of your choosing. Whether that be Ubuntu's **focal, jammy, impish, kinetic, etc...** Or whether you would like to use Debian's **buster, bullseye, bookworm, sid, etc...**
You can also add whatever other targets as well after the "-t" option, using a comma (no space) to separate each. Here's a few:

`sudo CROUTON_BRANCH=silence crouton -r xenial -t xfce,xfce-desktop,core,keyboard,xiwi,x11,cli-extra,extension`

**Comment by ClaudeJGreengrass** [comment]
Thanks @jake01756

I also can confirm
sudo CROUTON_BRANCH=silence crouton -r jammy -t xfce,xiwi
works for me.

**Comment by Mutmaheen001** [comment]
Thanks @jake01756 

The code worked

**Comment by mio-19** [comment]
> You could try this PR, second commit might fix it: [0d48e40](https://github.com/dnschneid/crouton/commit/0d48e401d7375724828adc448568c192f4d119dd)
> 
> (none of this is tested -- at all)

doesn't work

**Comment by GamerX999999999** [comment]
> Okay I found a temporary solution!
> 
> Use this command when invoking crouton to use the silence branch: `sudo CROUTON_BRANCH=silence crouton`
> 
> Ex: `sudo CROUTON_BRANCH=silence crouton -u -r bullseye -t xfce`
> 
> By the looks of things it appears that you will be without sound? I'm not quite sure. But hey it works.

I got this "/usr/local/bin/crash_reporter_wrapper: 119: exec: /var/run/crw/crw: Permission denied
WARNING: Unable to register core dump handler."

**Comment by Qwer-TeX** [comment]
> > Okay I found a temporary solution!
> > Use this command when invoking crouton to use the silence branch: `sudo CROUTON_BRANCH=silence crouton`
> > Ex: `sudo CROUTON_BRANCH=silence crouton -u -r bullseye -t xfce`
> > By the looks of things it appears that you will be without sound? I'm not quite sure. But hey it works.
> 
> I got this "/usr/local/bin/crash_reporter_wrapper: 119: exec: /var/run/crw/crw: Permission denied WARNING: Unable to register core dump handler."

I got the same error while installing focal, and hiresuite. but Jammy gave me no error.

**Comment by swettyboy69** [comment]
same issue here (`sed: can't read common/cras_messages.h: No such file or directory`)

**Comment by EPICTHUNDER123** [comment]
hey, same issue here and i am using a hp chromebook x360 with touchscreen. been looking for something that would work, jammy does work for my chromebook.

**Comment by EPICTHUNDER123** [comment]
update, it downloads but after it requires updates.

**Comment by EPICTHUNDER123** [comment]
still wont work now i getError: not connected.
Cannot connect to extension, retrying...
Error: not connected.
Cannot connect to extension, retrying...
Error: not connected.
Cannot connect to extension, retrying...
Error: not connected.
Cannot connect to extension, retrying...
Error: not connected.
Cannot connect to extension, retrying...
Error: not connected.
Cannot connect to extension, retrying...
Error: not connected.
Cannot connect to extension, retrying...
Error: not connected.
Cannot connect to extension, retrying...
Error: not connected.
Cannot connect to extension, retrying...
Error: not connected.
Cannot connect to extension, retrying...
Unable to start display, make sure the crouton extension is installed
and enabled, and up to date. (download from https://goo.gl/OVQOEt)
Running exit commands...
/usr/bin/xinit: connection to X server lost

waiting for X server to shut down Hangup
Hangup
Hangup
(II) Server terminated successfully (0). Closing log file.

Unmounting /mnt/stateful_partition/crouton/chroots/jammy...
Sending SIGTERM to processes under /mnt/stateful_partition/crouton/chroots/jammy...
chronos@localhost / $

**Comment by EPICTHUNDER123** [comment]
i get the error while launching xfce4

**Comment by Qwer-TeX** [comment]
> hey, same issue here and i am using a hp chromebook x360 with touchscreen. been looking for something that would work, jammy does work for my chromebook.

What a coincidence. I've been using the same chromebook as you and you might have the same error as me!!

**Comment by EPICTHUNDER123** [comment]
im using it on stable channel, enabled dev mode, version is Version 114.0.5735.143 (Official Build) (64-bit), I used the sudo CROUTON_BRANCH=silence crouton -r jammy -t xfce,xiwi command, could just be type of chromebook...

**Comment by sb294188** [comment]
I got past this by executing from /usr/local like this: sudo -cp ~/Downloads/crouton /usr/local, and then sudo sh /usr/local/crouton -kali-rolling.

**Comment by Qwer-TeX** [comment]
@EPICTHUNDER123 
Im using ChromeOs 114.0.5735.239. We both have dev mode enabled and are on the stable channel. Our versions are similar and aren't the reason of such error. 

Probably you didn't install the crouton browser extension. install it from [here](https://goo.gl/OVQOEt) I don't know what it fully does but it has some debugging and logging features.

if it doesn't work then type `sudo startxfce4` again, again, and again. And it worked in my case. I don't know the reason why it sometimes does that.

If it then doesn't work, then i I don't know what it could be. I know that its not the type of chromebook bc their software are mostly standard.

**Comment by EPICTHUNDER123** [comment]
I have the chrome extension but i should try spaming it, illl get back if it does not work :)

**Comment by EPICTHUNDER123** [comment]
so good news spamming worked once, now it jut shows green blue and red squares on the window...

**Comment by EPICTHUNDER123** [comment]
![Screenshot 2023-07-28 3 01 18 PM](https://github.com/dnschneid/crouton/assets/118402888/9d0242c0-75db-4ab7-9aa0-ceb358751525)

**Comment by EPICTHUNDER123** [comment]
all it says is


find_shm: Cannot connect to findnacl daemon. (Connection refused)
write_image: Cannot find shm, moving on...
find_shm: Cannot connect to findnacl daemon. (Connection refused)
write_image: Cannot find shm, moving on...
find_shm: Cannot connect to findnacl daemon. (Connection refused)
write_image: Cannot find shm, moving on...
find_shm: Cannot connect to findnacl daemon. (Connection refused)
write_image: Cannot find shm, moving on...
find_shm: Cannot connect to findnacl daemon. (Connection refused)
write_image: Cannot find shm, moving on...
find_shm: Cannot connect to findnacl daemon. (Connection refused)
write_image: Cannot find shm, moving on...
find_shm: Cannot connect to findnacl daemon. (Connection refused)
write_image: Cannot find shm, moving on...
find_shm: Cannot connect to findnacl daemon. (Connection refused)
write_image: Cannot find shm, moving on...
find_shm: Cannot connect to findnacl daemon. (Connection refused)
write_image: Cannot find shm, moving on...
find_shm: Cannot connect to findnacl daemon. (Connection refused)
write_image: Cannot find shm, moving on...
find_shm: Cannot connect to findnacl daemon. (Connection refused)
write_image: Cannot find shm, moving on...

**Comment by EPICTHUNDER123** [comment]
i've tried updating the ch root but it does nothing if you experience this then if possible tell me how to fix it!

**Comment by Qwer-TeX** [comment]
I see what happened, The startup script cannot find XFree86, and X11 used to show you the nice windows, menus, apps, icons, etc. if `sudo enter-chroot` works, and `sudo startxfce4` doesn't work, then it cannot find the X Window System. to fix that update the chroot to include the targets `x11`. I also suggest to install the `core` target because it contains core system configuration used to configure the X Window System. It should work, I guaranty it!

**Comment by NazagamerITA** [comment]
> Okay I found a temporary solution!
> 
> Use this command when invoking crouton to use the silence branch: `sudo CROUTON_BRANCH=silence crouton`
> 
> Ex: `sudo CROUTON_BRANCH=silence crouton -u -r bullseye -t xfce`
> 
> By the looks of things it appears that you will be without sound? I'm not quite sure. But hey it works.

Thanks! it worked!

**Comment by EPICTHUNDER123** [comment]
this worked thank you, also one more question, if you are installing lunix apps on xfce4 do you need to enable lunix on your chromebook?

**Comment by DennisLfromGA** [comment]
@EPICTHUNDER123,

No, the Crostini / Linux development environment is completely separate and not needed for crouton.

It is a good alternative to crouton for some and does not require Developer mode. 
You can check out crouton author's note about it here: https://github.com/dnschneid/crouton#what-about-dem-crostinis-though

-**DennisLfromGA**

**Comment by EPICTHUNDER123** [comment]
this was not what i was getting at, lets say you install minecraft on chromebook linux, would minecraft be on xfce4 too?

**Comment by DennisLfromGA** [comment]
@EPICTHUNDER123,

> if you are installing lunix apps on xfce4 do you need to enable lunix on your chromebook?

Again, no. They are isolated, Crostini is in a VM in a container, it has nothing to do with the crouton chroot.

> Lets say you install minecraft on chromebook linux, would minecraft be on xfce4 too?

No, see the above explanation.

-**DennisLfromGA**

**Comment by kiarog3** [comment]
I received the same error. Has anyone found another work around that doesn't affect the sound. I downloaded it using the unity interface.. and tried to start but received this message. I've also downloaded the crouton extension because somewhere said it would help.

**Comment by kiarog3** [comment]
also which interface is most preferred

**Comment by Qwer-TeX** [comment]
Xfce4 interface is most preferred because its lightweight and doesn't have eyecandy(animations) like unity. Usage of ram is about 354 MB for running crouton with xfce4 interface.

**Comment by brianwei20172218** [comment]
I ran into same problem...
I used to have positive feeling about chromebooks, but now gradually I feel it is trap :) 
Once you get it, it so hard to go anywhere.

**Comment by redd-rl** [comment]
Solved this issue personally by installing CRAS using chromebrew, "crew install cras" and then running crouton setup, knowledge only goes as far as ubuntu.

**Comment by redd-rl** [comment]
dash ls. not Ls, -Ls.

**Comment by redd-rl** [comment]
also i kinda lied i didn't get further than that i was forced to downgrade to chrome os 109 to advance, i installed unity image with targets xiwi,unity.

sudo CROUTON_BRANCH=silence crouton -t xiwi,unity
i think.
got me a barely working instance on chrome os 109, don't want to try updating to 121 again.
